routes/productRouter.js

```diff
@@ -0,0 +1,14 @@
+const express = require('express');
+const router = express.Router();
+
+const productController = require('../controllers/productController');
+
+router.get('/', productController.getProductsForMainPage);
+router.get('/products', productController.getProductsByMainCategory);
+router.get('/body-hand/hand', productController.getProductsForHands);
+router.get('/body-hand/body', productController.getProductsForBodys);
+router.get('/products/detail', productController.getProductById);
```

Comment:
index.js 파일을 통해 라우터가 이미 한차례 분기처리 된 이후의 리소스를 포함한 엔드포인트를 설계할 수 있습니다.

- 이미 `/products` 라는 path가 설정되어 있기 때문에 products를 한번 더 작성할 필요가 없습니다. 또한 카테고리 별 상품을 조회하는 것은 `필터링`을 통해 구현해볼 수 있을 것 같습니다.

- body-hand/hand 와 body-hand/body와 관련된 엔드포인트 설계는 다시 한번 고려해 볼 필요가 있어보입니다. 지금은 body-hand 제품만 포함하고 있지만, 더 많은 상품을 카테고리별로 엔드포인트를 구분한다면 이는 비효율적인 작업을 초래할 수도 있습니다. `body-hand`를 포함하고 있는 가장 큰 범주 (테이블 명)에 따른 id 값을 path-parameter로 조회하는 것은 어떨까요?

- 상품의 상세 페이지 내부에서 상세 정보를 확인할 수 있는 엔드포인트에서도 products라는 리소를 명시하는 것이 아니라 `productId`를 path parameter로 받도록 수정해주면 좋을 것 같습니다.

```suggestion
router.get('/', productController.getProductsForMainPage);
router.get('/body-hand/hand', productController.getProductsForHands);
router.get('/body-hand/body', productController.getProductsForBodys);
router.get('/detail/:productId', productController.getProductById);
```